src/Components/Radio/Radio.component.jsx

```diff
@@ -108,6 +124,21 @@ const Radio = ({activeTrack, setActiveTrack}) => {
     }
   }
 
+  useEffect( 
+    () => {
+      if (playing) {
+        //update on play, then every 10 seconds
+        fetchData();
+        const interval = setInterval(
+          () => {
+            fetchData(); 
+          }, 10000);
+        //cleanup effect
+        return () => clearInterval(interval);
+      }
+    }
+  );
```

Comment:
blocker

- without the inclusion of a dependency array this useEffect is going to run on every re-render which will spin up multiple intervals which is not the desired behavior. 

- See the array at the end of the useEffect on line 101? The array at the end includes the dependency activeTrack so it is only going to run when the activeTrack value changes.

Reply:
sold. again: dependency array did not like having fetchData() declared, solved by moving it into scope